src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -231,24 +212,22 @@ const TableHeader: React.FC<TableHeaderProps> = ({
         <div className="table-header__lock-indication">
           {isCurrentTableLocked ? "Table Locked" : "Lock Table"}
         </div>
-        <label className="switch table-header__lock-toggle">
-          <input
-            type="checkbox"
-            className="switch-hidden-input"
-            checked={!!isCurrentTableLocked}
-            onChange={toggleIsCurrentTableLocked}
-          />
-          <span className="slider" />
-        </label>
+        <Toggler
+          toggled={!!isCurrentTableLocked}
```

Comment:
We shouldn't be using `!!` to force this here. `isCurrentTableLocked` should be forced to be a `boolean` at the point it's defined.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/4b7532e6f02068126c212a6f0dab1714aaaad55a

Reply:
As an addition to the fix here, it might be good to explicitly set the return type on line 66'ish where `isCurrentTableLocked` to set it to `: boolean`, and probably also to use `isTruthy` in place of the `!!` there. It also seems to be `useMemo`'d, which I don't think is relevant here since it's just a `boolean`?

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/f6e11778a3edad587868e67cdb5bb100ff16473d